### PR TITLE
Increase maximum queue size and remove global configuration parameter

### PR DESCRIPTION
**Pull request by hanshuebner**
### Summary

- Make the default for the maximum number of batches queued in a plugin queue large to accommodate large installations.  
- Remove the global kong.conf parameter that allows the maximum number of batches to be set
- Add an undocumented, last-resort environment variable KONG_MAX_QUEUED_BATCHES that can be used to override the global default.

We don't want to expose a global configuration option for the queue limit to prevent us from having to support it in the future.  The queue system will be completely reworked in 3.2, hence this band-aid is only there as a bridge.

EE PR: https://github.com/Kong/kong-ee/pull/4541

### Checklist

- [X] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-303

### CODE REVIEW

File: kong/tools/batch_queue.lua
Comment (dndx) [inline kong/tools/batch_queue.lua:247]:
```suggestion
    else

```

File: kong/tools/batch_queue.lua
Comment (dndx) [inline kong/tools/batch_queue.lua:252]: Let's move this to the top of this file for visibility.

File: kong/tools/batch_queue.lua
Comment (dndx) [inline kong/tools/batch_queue.lua:253]:
```suggestion
  local max_queued_batches = opts.max_queued_batches or
        getenv_number("KONG_MAX_QUEUED_BATCHES", DEFAULT_MAX_QUEUED_BATCHES)
```

**Review by dndx — COMMENTED** [review]
This looks like a reasonable solution with a reasonable default. But after discussing with @kikito we have decided to only backport this into 3.2 for now. The rest of the backport decision will be made later since these are not being released anytime soon.

**Comment by hanshuebner** [comment]
@dndx I've cherry-picked the EE changes as those were already improved in response to reviews from @flrgh and @kikito - Your feedback should be covered by that.

**Review by dndx — APPROVED** [review]

**Comment by dndx** [comment]
@hanshuebner Thanks for the work. I did a little bit of style tweak but otherwise this PR is good to go. Just waiting for CI and need to cherry pick again from master to EE after it is merged.

**Pull request merged**
